File: API/Entities/Appointment.cs
using System;

namespace API.Entities
{
    public class Appointment
    {
        public int Id { get; set; }
        public DateTime StartsAt { get; set; }
        public DateTime EndsAt { get; set; }
        public int Duration { get; set; }
        public string Note { get; set; }

        public int? ClientId { get; set; }
        public Client Client { get; set; }
        public int BarberId { get; set; }
        public Barber Barber { get; set; }
        public int AppointmentTypeId { get; set; }
        public AppointmentType AppointmentType { get; set; }
        public int AppointmentStatusId { get; set; }
        public AppointmentStatus AppointmentStatus { get; set; }
    }
}

File: client/src/app/models/appointment.ts
import { AppointmentType } from './appointmentType';
import { Barber } from './barber';
import { Client } from './client';

export interface Appointment {
  id: number;
  startsAt: Date;
  endsAt: Date;
  duration: number;
  appointmentTypeId: number;
  barberId: number;
  clientId: number;
  appointmentType: AppointmentType;
  client: Client;
  barber: Barber;
  appointmentStatusId: number;
  note: string;
}
